File: 04-python/ordenamiento/03-seleccion.py
# input 
nums = [10,5,1,20,2,6]

def ordenamientoSeleccion(array):
    n = len(array)
    for i  in range(n):
        min = i 
        for j in range(i+1,n):
            if array[min] > array[j]:
                min = j
        # array[i], array[min] = array[min], array[i]
        temp = array[i]
        array[i] = array[min]
        array[min] = temp
    return array

print( ordenamientoSeleccion(nums))

File: 02-javascript/ordenamiento/03-seleccion.js
// Input 

let nums = [72,50,10,44,8,20];

function swap(arr, primerPosicion, segundaPosicion){
    [72,50,10,44,8,20]
    //primerPosicion = 0;
    let temp = arr[primerPosicion]; // 72
    arr[primerPosicion] = arr[segundaPosicion]; // 8
    arr[segundaPosicion] = temp;
}

function ordenamientoSeleccion( nums ){
    let i, j, min;
    for (i = 0; i < nums.length; i++) {
        min = i;
        for(j = i +1; j< nums.length; j++)
            if(nums[j]< nums[min])
                min = j;
        swap(nums,min,i);
    }
    return nums;
}

let demo = ordenamientoSeleccion(nums);

console.log(demo);